src/fourier.jl

```diff
@@ -0,0 +1,773 @@
+function positive_fft_bins(n_bin::Integer; include_zero::Bool = false)
+    minbin = 2
+    if include_zero
+        minbin = 1
+    end
+    return (minbin : (n_bin+1) ÷ 2)
+end
+
+function poisson_level(norm::String; meanrate = nothing, n_ph = nothing, backrate::Real = 0.0)
+    if norm == "abs"
+        return 2.0 * meanrate
+    elseif norm == "frac"
+        return 2.0 / (meanrate - backrate)^2 * meanrate
+    elseif norm == "leahy"
+        return 2.0
+    elseif norm == "none"
+        return float(n_ph)
+    else 
+        throw(ArgumentError("Unknown value for norm: $norm"))
+    end
+end
+
+function normalize_frac(unnorm_power::AbstractVector{<:Number}, dt::Real, n_bin::Integer, 
+                        mean_flux::Real; background_flux::Real=0.0)
+    if background_flux > 0
+        power = unnorm_power * 2 * dt / ((mean_flux - background_flux) ^ 2 *
+                                          n_bin)
+    else
+        # Note: this corresponds to eq. 3 in Uttley+14
+        power = unnorm_power * 2 * dt / (mean_flux ^ 2 * n_bin)
+    end
+    return power
+end
+
+normalize_abs(unnorm_power::AbstractVector{<:Number}, dt::Real, n_bin::Integer) = 
+    unnorm_power * 2 / n_bin / dt
+
+normalize_leahy_from_variance(unnorm_power::AbstractVector{<:Number}, 
+                              variance::Real, n_bin::Integer) = 
+    unnorm_power * 2 / (variance * n_bin)
+
+normalize_leahy_poisson(unnorm_power::AbstractVector{<:Number}, n_ph::Real) = 
+    unnorm_power * 2 / n_ph
+
+function normalize_periodograms(unnorm_power::AbstractVector{<:Number}, dt::Real, 
+                                n_bin::Integer; mean_flux=nothing, n_ph=nothing,
+                                variance=nothing, background_flux::Real=0.0, 
+                                norm::String="frac",power_type::String="all")
+    
+    if norm == "leahy" && !isnothing(variance)
+        pds = normalize_leahy_from_variance(unnorm_power, variance, n_bin)
+    elseif norm == "leahy"
+        pds = normalize_leahy_poisson(unnorm_power, n_ph)
+    elseif norm == "frac"
+        pds = normalize_frac(
+            unnorm_power, dt, n_bin, mean_flux,
+            background_flux=background_flux)
+    elseif norm == "abs"
+        pds = normalize_abs(unnorm_power, dt, n_bin)
+    elseif norm == "none"
+        pds = unnorm_power
+    else 
+        throw(ArgumentError("Unknown value for norm: $norm"))
+    end
+
+    if power_type == "all"
+        return pds
+    elseif power_type == "real"
+        return real(pds)
+    elseif power_type in ["abs", "absolute"]
+        return abs.(pds)
+    else 
+        throw(ArgumentError("Unknown value for power_type: $power_type"))
+    end
+
+end
+
+function bias_term(power1::Real, power2::Real, power1_noise::Real, 
+                   power2_noise::Real, n_ave::Integer;
+                   intrinsic_coherence::Real=1.0)
+    
+    if n_ave > 500
+        return 0.0 * power1
```

Comment:
```suggestion
        return float(zero(power1))
```

Reply:
Actually I should just return 0.0 now as the function is meant to be used with dot broadcast for vectors.